PhysicsTools/PatAlgos/python/producersHeavyIons/heavyIonJetSetup.py

```diff
@@ -0,0 +1,229 @@
+    genjetcollection = 'ak' + str(radius) + 'HiGenJets'
+
+    addToProcessAndTask(
+        genjetcollection,
+        ak5HiGenJets.clone(rParam = radius / 10),
+        process,
+        task)
+
+    addToProcessAndTask(
+        'ak' + str(radius) + 'HiCleanedGenJets',
+        heavyIonCleanedGenJets.clone(src = genjetcollection),
+        process,
+        task)
+
+    modules = {
+        'patJetGenJetMatch':
+        patJetGenJetMatch.clone(
+            matched = 'ak' + str(radius) + 'HiCleanedGenJets',
+            maxDeltaR = radius / 10,
+            resolveByMatchQuality = True,
+            src = tag + "Jets",
+            ),
+
+        'patJetPartonMatch':
+        patJetPartonMatch.clone(
+            matched = "cleanedPartons",
+            src = tag + "Jets",
+            ),
+
+        'patJetPartons':
+        patJetPartons.clone(
+            particles = "hiSignalGenParticles",
+            ),
+
+        'patJetFlavourAssociation':
+        patJetFlavourAssociation.clone(
+            jets = tag + "Jets",
+            rParam = radius / 10,
+            bHadrons = "patJetPartons" + tag + ":bHadrons",
+            cHadrons = "patJetPartons" + tag + ":cHadrons",
+            leptons = "patJetPartons" + tag + ":leptons",
+            partons = "patJetPartons" + tag + ":physicsPartons",
+            ),
+
+        'patJetPartonAssociationLegacy':
+        patJetPartonAssociationLegacy.clone(
+            jets = tag + "Jets",
+            partons = "allPartons",
+            ),
+
+        'patJetFlavourAssociationLegacy':
+        patJetFlavourAssociationLegacy.clone(
+            srcByReference = "patJetPartonAssociationLegacy" + tag,
+            ),
+        }
+
+    for label, module in six.iteritems(modules):
+        addToProcessAndTask(label + tag, module, process, task)
+
+def useCsJetsForPat(process, tag):
+    process.patJets.useLegacyJetMCFlavour = True
+    process.patJets.userData.userFloats.src = []
+    process.patJets.userData.userInts.src = []
+    process.patJets.jetSource = tag + "Jets"
+    process.patJets.genJetMatch = "patJetGenJetMatch" + tag
+    process.patJets.genPartonMatch = "patJetPartonMatch" + tag
+    process.patJets.JetFlavourInfoSource = "patJetFlavourAssociation" + tag
+    process.patJets.JetPartonMapSource = "patJetFlavourAssociationLegacy" + tag
+    process.patJets.jetCorrFactorsSource = ["patJetCorrFactors" + tag]
+    process.patJets.trackAssociationSource = "JetTracksAssociatorAtVertex" + tag
+    process.patJets.useLegacyJetMCFlavour = True
+    process.patJets.discriminatorSources = [
+        "SimpleSecondaryVertexHighEffBJetTags" + tag,    
+        "SimpleSecondaryVertexHighPurBJetTags" + tag,
+        "CombinedSecondaryVertexBJetTags" + tag,
+        "CombinedSecondaryVertexV2BJetTags" + tag,
+        "JetBProbabilityBJetTags" + tag,
+        "JetProbabilityBJetTags" + tag,
+        "TrackCountingHighEffBJetTags" + tag,
+        "TrackCountingHighPurBJetTags" + tag,
```

Comment:
none of these are available in the HI AOD.
IIRC, all of these b-taggers were dropped in pp since start of run2.
Are these  still supported/used in the HI context, even in run3?

Please point me to a reference documentation or recipe page which was used to derive updates in this PR. 
I noticed that the addJetCollection-based variant had 2 less tags; so, it's not obvious if all were really needed.

Reply:
The analysis setup for 2018 that we are trying to reproduce in miniAOD may be found here:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/HiForestSetup#Setup_for_10_3_X_2018_PbPb_data
For Run 2, we are still using the b-tagging setup that pp used in Run 1.
For Run 3, there is no reason to continue doing this.  We will switch to candidate based tagging, add some the newer taggers, and remove some of the deprecated ones.  Let's do that in a separate PR, after we're done with the Run 2 miniAOD setup.   The scope of the current miniAOD campaign is to reproduce exactly our "legacy" setup.
We also have an legacy b-tagging calibration based on this version of csvV2, that I will need to include in a subsequent PR.  
IIRC the addJetCollection didn't allow for two of the taggers, so I was forced to drop them.  
The only taggers strictly needed are csvV2 and JetProbability.  The rest are occasionally useful for spot checks, but they could be dropped, if necessary.